pkg/vmprovider/providers/vsphere2/network/network.go

```diff
@@ -518,6 +521,155 @@ func ncpNetIfToResult(
 	return result, nil
 }
 
+// VPCCRName returns the name to be used for the VPC SubnetPort CR.
+func VPCCRName(vmName, networkName, interfaceName string) string {
+	var name string
+
+	if networkName != "" {
+		name = fmt.Sprintf("%s-%s-%s", vmName, networkName, interfaceName)
+	} else {
+		name = fmt.Sprintf("%s-%s", vmName, interfaceName)
+	}
+
+	return name
+}
+
+func createVPCNetworkInterface(
+	vmCtx context.VirtualMachineContextA2,
+	client ctrlruntime.Client,
+	vimClient *vim25.Client,
+	clusterMoRef *vimtypes.ManagedObjectReference,
+	interfaceSpec *vmopv1.VirtualMachineNetworkInterfaceSpec) (*NetworkInterfaceResult, error) {
+
+	kind := interfaceSpec.Network.Kind
+	if kind != "" && kind != "SubnetSet" && kind != "Subnet" {
+		return nil, fmt.Errorf("network kind %q is not supported for VPC", kind)
+	}
+
+	networkName := interfaceSpec.Network.Name
+	vpcSubnetPort := &vpcv1alpha1.SubnetPort{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      VPCCRName(vmCtx.VM.Name, networkName, interfaceSpec.Name),
+			Namespace: vmCtx.VM.Namespace,
+		},
+	}
+
+	if kind == "SubnetSet" {
```

Comment:
Since L545 allows `kind == ""` we should handle that here too (I think they want SubnetSet to be the default but double check). And kind of related, if neither `SubnetSet` or `Subnet` is set because `networkName == ""`, does VPC behave like NCP and fallback to the NS default?

A switch statement is probably more readable here too.